pulpcore/pulpcore/app/serializers/repository.py

```diff
@@ -249,6 +250,28 @@ class Meta:
             'name', 'base_path', 'publisher', 'publication', 'base_url', 'repository',
         )
 
+    def validate_base_path(self, path):
+        # look for any base paths nested in path
+        search = path.split("/")[0]
+        q = Q(base_path=search)
+        for subdir in path.split("/")[1:]:
+            search = "/".join((search, subdir))
+            q |= Q(base_path=search)
+
+        # look for any base paths that nest path
+        q |= Q(base_path__startswith='{}/'.format(path))
+        qs = models.Distribution.objects.filter(q)
+
+        if self.instance is not None:
+            qs = qs.exclude(pk=self.instance.pk)
+
+        match = qs.first()
+        if match:
+            raise serializers.ValidationError(detail=_("overlaps with existing distributor "
+                                                       "{}").format(match.name))
```

Comment:
Please capitalize "overlaps" and put single quotes around the name of the distributor to distinquish it e.g.

```Overlaps with existing distrubutor 'bar'```